File: puzzles/easy/power_of_thor_episode_1/main.java
import java.util.*;

class Player {
    public static void main(String[] args) {
        @SuppressWarnings("resource")
        Scanner in = new Scanner(System.in);

        int lightX = in.nextInt();
        int lightY = in.nextInt();
        int initialTX = in.nextInt();
        int initialTY = in.nextInt();

        int positionX = initialTX;
        int positionY = initialTY;

        while (true) {
            @SuppressWarnings("unused")
            int remainingTurns = in.nextInt();

            String direction = "";

            if (positionY < lightY) {
                direction += "S";
                positionY++;
            }
            if (positionY > lightY) {
                direction += "N";
                positionY--;
            }
            if (positionX < lightX) {
                direction += "E";
                positionX++;
            }
            if (positionX > lightX) {
                direction += "W";
                positionX--;
            }
            System.out.println(direction);
        }
    }
}

File: puzzles/easy/power_of_thor_episode_1/c/main.c
#include <stdio.h>

int light_x;
int light_y;
int initial_tx;
int initial_ty;

void solve();

int main() {
    scanf("%d%d%d%d", &light_x, &light_y, &initial_tx, &initial_ty);
    solve();

    return 0;
}

void solve() {
    while (1) {
        int remianing_turns;
        scanf("%d", &remianing_turns);

        if (initial_ty < light_y) {
            printf("S");
            initial_ty++;
        }
        if (initial_ty > light_y) {
            printf("N");
            initial_ty--;
        }
        if (initial_tx < light_x) {
            printf("E");
            initial_tx++;
        }
        if (initial_tx > light_x) {
            printf("W");
            initial_tx--;
        }
        printf("\n");
    }
}
